course_content/notebooks/dask_intro.ipynb

```diff
@@ -0,0 +1,216 @@
+   "metadata": {},
+   "source": [
+    "#### Scheduler\n",
+    "One of Dask’s key benefits is its ability to efficiently schedule tasks to optimize computations. Everything Dask does is built on top of “schedulers”. These schedulers take the order of work established by a task graph and find the optimal way to break down and carry out the tasks.\n",
+    "\n",
+    "Dask has four types of schedulers:\n",
+    "- Synchronous: Single thread (good for debugging).\n",
+    "        dask.get\n",
+    "- Threaded: Utilises a thread pool.\n",
+    "        dask.threaded.get\n",
+    "- Multiprocessing: Utilises a process pool.\n",
+    "        dask.multiprocessing.get\n",
+    "- Distributed: Utilises a cluster of distributed machines.\n",
+    "        distributed.Client.get\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Worker\n",
+    "A worker receives work from the scheduler to process and returns output to the scheduler."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Graph\n",
+    "A way to represent 'things' and their relationships where a node (circle) is a thing and an edge (line) between nodes is their relationship e.g. a social network where nodes are people and edges are their friendships.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Task Graph\n",
+    "A graph of tasks (nodes) and the data which is required to pass between them (edges).\n",
+    "In the context of dask:\n",
+    "- Task = circle\n",
+    "- Data = box\n",
+    "- Direction of flow = arrow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def add(a, b):\n",
+    "    return a + b\n",
+    "\n",
+    "x = 1\n",
+    "y = 2\n",
+    "z = add(x, y)\n",
+    "w = sum([x, y, z])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![](../images/task_graph_def_img_1.png)\n",
+    "Viewing the relationship between tasks and data can reveal better\\* ways to order the tasks to achieve the same computation.\n",
+    "\n",
+    "\\* Better can mean in less time or using less memory or both!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Dask Graph\n",
+    "Dask stores task graphs in a Python dictionary which maps keys to computations:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "dsk = {'x': 1,\n",
+    "       'y': 2,\n",
+    "       'z': (add, 'x', 'y'),\n",
+    "       'w': (sum, ['x', 'y', 'z'])\n",
+    "      }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dask is different from other parallelising libraries in that it uses ordinary Python structures instead of a specialised API:\n",
```

Comment:
For improved clarity, how about:

"...uses ordinary Python structures _to define processing tasks_ instead of..."

Reply:
How about:
"Dask is different from other parallelising libraries in that it uses ordinary Python structures to represent task graphs instead of a specialised API" ?